staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter_test.go

```diff
@@ -0,0 +1,166 @@
+				LendablePercent:          &lendable,
+				BorrowingLimitPercent:    &borrowingLimit,
+				LimitResponse: flowcontrol.LimitResponse{
+					Type: flowcontrol.LimitResponseTypeQueue,
+					Queuing: &flowcontrol.QueuingConfiguration{
+						Queues:           10,
+						HandSize:         2,
+						QueueLengthLimit: 10,
+					},
+				},
+			},
+		},
+	}
+	cfgObjs = append(cfgObjs, fsObj, plcObj)
+
+	clientset := clientsetfake.NewSimpleClientset(cfgObjs...)
+	informerFactory := informers.NewSharedInformerFactory(clientset, time.Second)
+	flowcontrolClient := clientset.FlowcontrolV1beta3()
+	clk := eventclock.Real{}
+	controller := newTestableController(TestableConfig{
+		Name:                   "Controller",
+		Clock:                  clk,
+		AsFieldManager:         ConfigConsumerAsFieldManager,
+		FoundToDangling:        func(found bool) bool { return !found },
+		InformerFactory:        informerFactory,
+		FlowcontrolClient:      flowcontrolClient,
+		ServerConcurrencyLimit: 24,
+		RequestWaitLimit:       time.Minute,
+		ReqsGaugeVec:           metrics.PriorityLevelConcurrencyGaugeVec,
+		ExecSeatsGaugeVec:      metrics.PriorityLevelExecutionSeatsGaugeVec,
+		QueueSetFactory:        fqs.NewQueueSetFactory(clk),
+	})
+
+	informerFactory.Start(nil)
+
+	status := informerFactory.WaitForCacheSync(nil)
+	if names := unsynced(status); len(names) > 0 {
+		t.Fatalf("WaitForCacheSync did not successfully complete, resources=%#v", names)
+	}
+
+	go func() {
+		controller.Run(nil)
+	}()
+
+	// ensure that the controller has run its first loop.
+	err := wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (done bool, err error) {
+		return controller.hasPriorityLevelState(plcObj.Name), nil
+	})
+	if err != nil {
+		t.Errorf("expected the controller to reconcile the priority level configuration object: %s, error: %s", plcObj.Name, err)
+	}
+
+	reqInfo := &request.RequestInfo{
+		IsResourceRequest: false,
+		Path:              "/foobar",
+		Verb:              "GET",
+	}
+	noteFn := func(fs *flowcontrol.FlowSchema, plc *flowcontrol.PriorityLevelConfiguration, fd string) {}
+	workEstr := func() fcrequest.WorkEstimate { return fcrequest.WorkEstimate{InitialSeats: 1} }
+
+	flowUser := testUser{name: "test-user"}
+	rd := RequestDigest{
+		RequestInfo: reqInfo,
+		User:        flowUser,
+	}
+
+	// QueueNoteFn executes twice, once leaving the queue and once exiting the queue, which
+	// should result in ~1s wait time in the latency tracker
+	qnf := fq.QueueNoteFn(func(bool) { time.Sleep(500 * time.Millisecond) })
+	ctx := request.WithLatencyTrackers(context.Background())
+	controller.Handle(ctx, rd, noteFn, workEstr, qnf, func() {})
+
+	latencyTracker, ok := request.LatencyTrackersFrom(ctx)
+	if !ok {
+		t.Fatalf("error getting latency tracker: %v", err)
+	}
+
+	// latency tracker should report ~ 1s wait time with some wiggle room
+	latency := latencyTracker.PriorityAndFairnessTracker.GetLatency()
+	if latency.Seconds() < 0.99 || latency.Seconds() > 1.01 {
```

Comment:
Ran this a few times to check it won't flake:
```
~/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver$ go test ./pkg/util/flowcontrol -c -race -count 1
~/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver$ stress ./flowcontrol.test -test.run TestQueueWaitTimeLatencyTracker
5s: 36 runs so far, 0 failures
10s: 84 runs so far, 0 failures
15s: 132 runs so far, 0 failures
20s: 170 runs so far, 0 failures
25s: 217 runs so far, 0 failures
30s: 264 runs so far, 0 failures
35s: 312 runs so far, 0 failures
40s: 360 runs so far, 0 failures
45s: 400 runs so far, 0 failures
50s: 446 runs so far, 0 failures
55s: 493 runs so far, 0 failures
1m0s: 540 runs so far, 0 failures
1m5s: 588 runs so far, 0 failures
1m10s: 628 runs so far, 0 failures
1m15s: 675 runs so far, 0 failures
1m20s: 722 runs so far, 0 failures
1m25s: 769 runs so far, 0 failures
1m30s: 816 runs so far, 0 failures
1m35s: 864 runs so far, 0 failures
1m40s: 903 runs so far, 0 failures
1m45s: 950 runs so far, 0 failures
1m50s: 998 runs so far, 0 failures
1m55s: 1044 runs so far, 0 failures
2m0s: 1092 runs so far, 0 failures
2m5s: 1132 runs so far, 0 failures
2m10s: 1178 runs so far, 0 failures
2m15s: 1226 runs so far, 0 failures
2m20s: 1273 runs so far, 0 failures
2m25s: 1320 runs so far, 0 failures
2m30s: 1361 runs so far, 0 failures
2m35s: 1408 runs so far, 0 failures
2m40s: 1454 runs so far, 0 failures
2m45s: 1502 runs so far, 0 failures

```

Reply:
I'd still back it off to +/- 5% tolerance instead of +/- 1%

Reply:
Updated to +/- 5% (95, 105 ms)